src/App.tsx

```diff
@@ -1,24 +1,239 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  completeTodo,
+  editTodo,
+} from './api/todos';
+import { Filter } from './types/Filter';
+import { ErrorTypes } from './types/ErrorTypes';
+import {
+  Header, TodosList, Footer, ErrorNotifications,
+} from './components';
 
-const USER_ID = 0;
+const USER_ID = 6335;
 
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorTypes | null>(null);
+  const [filter, setFilter] = useState(Filter.All);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedTodoIds, setSelectedTodoIds] = useState<number[]>([]);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const pushError = (erroType: ErrorTypes, time = 3000) => {
+    setError(erroType);
+    setTimeout(() => setError(null), time);
```

Comment:
You should clear timeout on the component unmount
https://bobbyhadz.com/blog/react-cleartimeout

Reply:
@Anastasiia-Svintsova Thanks, but I need a little hint) How can I clear timeout created in parent element on unmounting of child one? should I pass to the state timout ID and then pass it to the props of child component every time I create timeout and in useEffect Hook in child component clear it?